pkg/pagerduty/pagerduty.go

```diff
@@ -2,112 +2,162 @@ package pagerduty
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"strings"
 	"time"
 
-	"github.com/PagerDuty/go-pagerduty"
+	pdApi "github.com/PagerDuty/go-pagerduty"
 )
 
-// PagerDutyClient defines the methods needed from the PagerDuty client.
-type PagerDutyClient interface {
-	GetClusterIDFromAlertList(alertList *pagerduty.ListAlertsResponse) (string, error)
-	GetClusterIDFromAlert(alert *pagerduty.IncidentAlert) (string, error)
-	GetClusterID(pdClient *pagerduty.Client, incidentID string) (string, error)
-	CreateIncident(description string) (string, error)
+// Alert struct represents the data contained in an alert.
+type Alert struct {
+	ID          string
+	Name        string
+	IncidentID  string
+	Severity    string
+	Status      string
+	CreatedAt   time.Time
+	WebURL      string
+	ClusterID   string
+	ClusterName string
 }
 
-// RealPagerDutyClient implements the PagerDutyClient interface using the real PagerDuty client.
-type RealPagerDutyClient struct {
-	PdClient *pagerduty.Client
+const (
```

Comment:
Those consts are not used anywhere?

Reply:
Yeah .. I used some intests and removed and unused once